tools/spectral/ipa/rulesets/functions/eachCustomMethodMustBeGetOrPost.js

```diff
@@ -0,0 +1,29 @@
+import { isCustomMethod } from './utils/resourceEvaluation.js';
+
+const ERROR_MESSAGE = 'The HTTP method for custom methods must be GET or POST.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+const VALID_METHODS = ['get', 'post'];
+const HTTP_METHODS = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch', 'trace'];
```

Comment:
[Just a note for future rules]: I expect we will have other similar rules coming up, at that point we can consider refactoring out a common method to check for valid/invalid methods to the utils package